pkg/gadget-service/service.go

```diff
@@ -62,16 +64,27 @@ type Service struct {
 	logger            logger.Logger
 	servers           map[*grpc.Server]struct{}
 	eventBufferLength uint64
+
+	// operatorGlobalParams stores all global parameters for DataOperators (non-legacy)
+	operators map[operators.DataOperator]*params.Params
 }
 
-func NewService(defaultLogger logger.Logger, length uint64) *Service {
+func NewService(defaultLogger logger.Logger) *Service {
+	ops := make(map[operators.DataOperator]*params.Params)
+	for _, op := range operators.GetDataOperators() {
+		ops[op] = apihelpers.ToParamDescs(op.GlobalParams()).ToParams()
+	}
 	return &Service{
-		servers:           map[*grpc.Server]struct{}{},
-		logger:            defaultLogger,
-		eventBufferLength: length,
+		servers:   map[*grpc.Server]struct{}{},
+		logger:    defaultLogger,
+		operators: ops,
 	}
 }
 
+func (s *Service) SetEventBufferLength(val uint64) {
+	s.eventBufferLength = val
+}
+
```

Comment:
nit: I suppose this change really belongs to the "cmd/ig: expose operator global params to cobra" commit.

I think this should also be together with commits that use it, otherwise the compilation is broken.

Reply:
I really need to find a better way to manage these kind of things - took way too long 😞